### PR TITLE
Ring spec compat

### PR DESCRIPTION
**Pull request by jcf**
Hi @weavejester!

I ran into the same issue reported by @malcolmsparks in #10, so figured I'd take a stab at fixing it.

This could be four PRs in theory as the protocol change, omitting query string, adding Clojure 1.8, and a 1.9 test are all reasonably self-contained. I wasn't sure how you'd prefer to receive the changes…

Let me know your thoughts.

### CODE REVIEW

File: test/ring/mock/request_test.clj [hunk 1]
```diff
@@ -131,3 +135,24 @@
...
+            false))
+    `(do
+       (require 'ring.core.spec)
+       ~@body)))
+
+(deftest test-specs
+  (when-clojure-spec
+    (doseq [req [(-> (request :get "/foo/bar") (query-string nil))
+                   (request :put "/foo/bar")
+                   (request :something "/foo" {:params true})]]
```
Comment (jcf) [inline test/ring/mock/request_test.clj]: Alignment Is off here. Clojure mode is really struggling with the indentation around `when-clojure-spec`… :disappointed:

**Comment by jcf** [comment]
Ping @weavejester. Any feedback on this one?

**Comment by weavejester** [comment]
Hi, sorry for the delayed reply. I've been a little swamped, recently.

I think rather than having `remove-nil-vals`, we should instead only add `:query-string` to the map if it is not `nil`. Otherwise their could be potential issues with keys that shouldn't be removed.

**Comment by jcf** [comment]
@weavejester I originally went with a `cond->` to associate the `:query-string` in but wasn't sure which Clojure versions were supported (Ring goes back to Clojure 1.3 I think?).

I did think it probably makes sense to get rid of keys with an explicit `nil` value because it seems wasteful, and semantically equivalent in this case, but will happily go the more conservative/safer route of only omitting `:query-string`.

**Comment by jcf** [comment]
Done! We now only remove `:query-string` from request maps.

**Comment by weavejester** [comment]
Ring goes back to 1.5.1, so `cond->` is supported (it was introduced in 1.5).

The reason I didn't want all `nil` values omitted is because someone could accidentally set the request method or URI to `nil`, and omitting that from the request map might make their error harder for them to catch.

**Comment by jcf** [comment]
Thanks, Mr. Reeves!

**Pull request merged**